src/components/Chats/ChaItem.tsx

```diff
@@ -0,0 +1,137 @@
+import { useParams } from "react-router-dom"
+import { io, Socket } from "socket.io-client";
+import { useMessagesQuery, useSendMessageQuery } from "../../hooks/useMessagesQuery";
+import { getAccessToken } from "../helpers/functions/getTokens";
+import { useEffect, useMemo, useRef, useState } from "react";
+import { handleConnect, handleDisconnect, socketUrl } from "../../services/socket";
+import { IMessageDTO } from "../../interfaces-submodule/interfaces/dto/message/imessage-dto";
+import { Loader } from "../Loader";
+import sprite from "../../images/svg/sprite.svg";
+import { useTheme } from "@emotion/react";
+import { Button, CustomTextarea, Form, FormWrap, Li, } from "../helpers/styles/ChatItem.styled";
+import { NotificationEvents } from "../../interfaces-submodule/enums/notification/notification-events.enum";
+
+export const ChatItem = () => {
+    const { id } = useParams<{ id: string }>();
+    const [content, setContent] = useState<string>("");
+    const [isFetching, setIsFetching] = useState<boolean>(false);
+    const [myMessage, setMyMessage] = useState<string>('')
+    
+    const { data, refetch, isLoading } = useMessagesQuery(id as string);
+    const { mutate: sendMessage } = useSendMessageQuery();
+
+    const accessToken = getAccessToken();
+    const theme: any = useTheme();
+    const chatListRef = useRef<HTMLUListElement>(null);
+
+    const chatsData: IMessageDTO[] = useMemo(() => {
+        return data || []
+    }, [data]);
+    
+    useEffect(() => {
+        if (chatListRef.current) {
+            chatListRef.current.scrollTop = chatListRef.current.scrollHeight;
+        }        
+    }, [chatsData, isFetching]);
+
+    useEffect(() => {
+        setIsFetching(false);
+        setMyMessage('');
+    },[chatsData])
+
+    useEffect(() => {
+        const socket: Socket = io(socketUrl, {
+            transports: ["websocket"],
+            timeout: 10000,
+        });
+
+        const requestBody = {
+            chatId: parseInt(id || '0'),
+            accessToken
+        };
+
+        socket.on('connect', () => handleConnect(socket, requestBody));
+        socket.on('disconnect', () => handleDisconnect(socket, requestBody));
+        socket.on(NotificationEvents.ChatResponse, () => {
```

Comment:
- [ ] on NotificationEvents.ChatResponse you just need to add response message to the chat list, but not refetch